android_app/app/src/main/java/com/health/openscale/core/bluetooth/BluetoothStandardWeightProfile.java

```diff
@@ -823,29 +825,59 @@ protected void chooseExistingScaleUser(Vector<ScaleUser> userList) {
         chooseScaleUserUi(choices);
     }
 
-    protected String getInitials(String fullName) {
-        if (fullName == null || fullName.isEmpty() || fullName.chars().allMatch(Character::isWhitespace)) {
-            return getDefaultInitials();
-        }
-        return buildInitialsStringFrom(fullName).toUpperCase();
+    protected String getInitials(final String fullName, final int size) {
+        if (0 == size)
+            return StringUtils.EMPTY;
+
```

Comment:
I think this test is redundant. Your `getDefaultInitials()` and  `buildInitialsStringFrom()` correctly generate empty strings if requested size is zero.
```suggestion
```

Reply:
true. However, this will reduce calls, stack creation and also if logic changes at some point it is a useful check to not break sw. Therefore, I would keep it.

Reply:
This code gets called rarely and is not part or any loop or recursion - why optimize for speed or stack?